src/Spago/Config.purs

```diff
@@ -334,18 +439,26 @@ readWorkspace { maybeSelectedPackage, pureBuild, migrateConfig } = do
             logDebug "Lockfile is up to date, using it"
             pure (Right contents)
 
-  -- Read in the package database
+loadPackageSet
+  :: ∀ a
+   . { workspaceConfig :: Core.WorkspaceConfig
+     , loadedPackages :: Map PackageName WorkspacePackage
+     , rootPath :: RootPath
+     , lockfile :: Either String Lockfile
+     }
+  -> Spago (Registry.RegistryEnv a) { packageSet :: PackageSet, compiler :: Range }
+loadPackageSet { lockfile, workspaceConfig, loadedPackages, rootPath } = do
   { offline } <- ask
-  packageSetInfo <- case maybeLockfileContents, workspace.packageSet of
+  packageSetInfo <- case lockfile, workspaceConfig.packageSet of
     _, Nothing -> do
       logDebug "Did not find a package set in your config, using Registry solver"
       pure Nothing
 
     -- If there's a lockfile we don't attempt to fetch the package set from the registry
     -- repo nor from the internet, since we already have the whole set right there
-    Right lockfile, _ -> do
+    Right lf, _ -> do
```

Comment:
(sorry, low-letter-count variables trip my brain)
```suggestion
    Right lockfile, _ -> do
```

Reply:
And this was due to potential shadowing of the parameter that you asked to rename to `eitherLockfile`

Reply:
Renamed.